File: 53_Maximum_Subarray/Solution.cpp
#include <iostream>
using namespace std;
#include <vector>

class Solution
{
public:
    int maxSubArray(vector<int> &nums)
    {
        int sum = 0;
        int maxsum = INT_MIN;
        for (auto num : nums)
        {
            sum += num;
            maxsum = max(sum, maxsum);
            if (sum < 0)
                sum = 0;
        }
        return maxsum;
    }
};

File: 53_Maximum_Subarray/Solution.java
class Solution {
    public int maxSubArray(int[] nums) {
        int max_sum = nums[0];
        int cur_sum = 0;
        for (int i = 0; i < nums.length; i++) {
            cur_sum += nums[i];
            if (cur_sum > max_sum) {
                max_sum = cur_sum;
            }
            if (cur_sum < 0) {
                cur_sum = 0;
            }

        }
        return max_sum;

    }
}